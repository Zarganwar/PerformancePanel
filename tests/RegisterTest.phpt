<?php

/**
 * Test: Register.
 *
 * @author Martin Jirásek <jertin@seznam.cz>
 */
namespace RegisterTest;
use Tester;
use Tester\Assert;
use \Zarganwar\PerformancePanel\Register;
require_once __DIR__ . '/bootstrap.php';
/**
 * @author Martin Jirásek <jertin@seznam.cz>
 * @testCase
 */
class RegisterTest extends Tester\TestCase
{
	protected function tearDown() {
		parent::tearDown();
		
	}
	
	public function testAddBreakpointNamed()
	{
		Register::addBreakpoint('A1_-=');
		$names = Register::getNames();
		Assert::equal(array('A1_-=' => 'A1_-='), $names);
	}
	public function testAddBreakpointNamedDuplicity()
	{
		Register::addBreakpoint('A1_-=');
		Register::addBreakpoint('A1_-=');
		$names = Register::getNames();
		Assert::equal(array('A1_-=', 'A1_-=_1'), $names);
	}
	public function testAddBreakpointUnnamed()
	{
		Register::addBreakpoint();
		Register::addBreakpoint();
		$names = Register::getNames();
		Assert::equal(array('BP_1', 'BP_2'), $names);
	}
}
\run(new RegisterTest());