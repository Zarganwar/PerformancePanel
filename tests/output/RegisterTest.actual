array('A1_-=' => 'A1_-=', 'A1_-=_2' => 'A1_-=_2', 'A1_-=_3' => 'A1_-=_3')