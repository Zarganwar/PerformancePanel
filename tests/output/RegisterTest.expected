array('A1_-=', 'A1_-=_1')